In EI 6.6.0 the VFS SMB2 fails on first attempt and successful only in second attempt.

### Description

VFS sender throws the below exception on the first attempt to write the file to windows file system. But after 30s later the file is created successfully in the target location.

> According to exception it fails to get the file type information from the windows file share for the first time, but according to VFSSender logic retry happens in 30 seconds and it was written successfully to windows file share.

```

TID: [-1234] [] [2022-11-25 10:04:27,506] ERROR {org.apache.synapse.transport.vfs.VFSTransportSender}

cannot resolve replyFile org.apache.commons.vfs2.FileSystemException:
Could not determine the type of file "smb2://domoff%2elocal;svc%5fp%5fsais%5fesb:***@spscs092.domoff.local/ENT_01$/SGC-Betrieb/Report/Input_CB_STATE_SMS".
at org.apache.commons.vfs2.provider.AbstractFileObject.getType(AbstractFileObject.java:1302)
at org.apache.commons.vfs2.provider.AbstractFileObject.exists(AbstractFileObject.java:900)
at org.apache.synapse.transport.vfs.VFSTransportSender.writeFile(VFSTransportSender.java:241)
at org.apache.synapse.transport.vfs.VFSTransportSender.sendMessage(VFSTransportSender.java:195)
at org.apache.axis2.transport.base.AbstractTransportSender.invoke(AbstractTransportSender.java:112)
at org.apache.axis2.engine.AxisEngine.send(AxisEngine.java:442)
at org.apache.axis2.description.OutOnlyAxisOperationClient.executeImpl(OutOnlyAxisOperation.java:297)
at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
at org.apache.synapse.core.axis2.Axis2FlexibleMEPClient.send(Axis2FlexibleMEPClient.java:640)
at org.apache.synapse.core.axis2.Axis2Sender.sendOn(Axis2Sender.java:85)
at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.send(Axis2SynapseEnvironment.java:572)
at org.apache.synapse.mediators.builtin.CallMediator.handleNonBlockingCall(CallMediator.java:279)
at org.apache.synapse.mediators.builtin.CallMediator.mediate(CallMediator.java:125)
at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:109)
at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:71)
at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
at org.apache.synapse.core.axis2.ProxyServiceMessageReceiver.receive(ProxyServiceMessageReceiver.java:225)
at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
at org.apache.axis2.transport.base.AbstractTransportListener.handleIncomingMessage(AbstractTransportListener.java:328)
at org.apache.axis2.transport.jms.JMSMessageReceiver.processThoughEngine(JMSMessageReceiver.java:214)
at org.apache.axis2.transport.jms.JMSMessageReceiver.onMessage(JMSMessageReceiver.java:124)
at org.apache.axis2.transport.jms.ServiceTaskManager$MessageListenerTask.handleMessage(ServiceTaskManager.java:702)
at org.apache.axis2.transport.jms.ServiceTaskManager$MessageListenerTask.run(ServiceTaskManager.java:531)
at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:829)Caused by: org.apache.commons.vfs2.FileSystemException: Unknown message with code "Could not get information for file: SGC-Betrieb\Report\Input_CB_STATE_SMS".
at org.apache.commons.vfs2.provider.smb2.Smb2ClientWrapper.getFileInfo(Smb2ClientWrapper.java:144)
at org.apache.commons.vfs2.provider.smb2.Smb2FileObject.getFileInfo(Smb2FileObject.java:112)
at org.apache.commons.vfs2.provider.smb2.Smb2FileObject.doGetType(Smb2FileObject.java:88)
```

### Steps to Reproduce

N/A

### Affected Component

MI

### Version

EI 6.6.0

### Environment Details (with versions)

_No response_

### Relevant Log Output

_No response_

### Related Issues

_No response_

### Suggested Labels

_No response_

--- a/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/AbstractOriginatingFileProvider.java
+++ b/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/AbstractOriginatingFileProvider.java
@@ -21,6 +21,7 @@ import org.apache.commons.vfs2.FileObject;
 import org.apache.commons.vfs2.FileSystem;
 import org.apache.commons.vfs2.FileSystemException;
 import org.apache.commons.vfs2.FileSystemOptions;
+import org.apache.commons.vfs2.impl.DefaultFileSystemManager;
 
 import java.util.Map;
 
@@ -108,11 +109,19 @@ public abstract class AbstractOriginatingFileProvider extends AbstractFileProvid
         // Check in the cache for the file system
         final FileName rootName = getContext().getFileSystemManager().resolveName(name, FileName.ROOT_PATH);
 
-        final FileSystem fs = getFileSystem(rootName, fileSystemOptions, defaultTimeout);
+        FileSystem fs = getFileSystem(rootName, fileSystemOptions, defaultTimeout);
+        FileObject testFile = fs.resolveFile(name);
+        try {
+            testFile.exists();
+        } catch (FileSystemException e) {
+            ((DefaultFileSystemManager) getContext().getFileSystemManager()).closeCachedFileSystem(name.getURI(), fileSystemOptions);
+            fs = getFileSystem(rootName, fileSystemOptions, defaultTimeout);
+            testFile = fs.resolveFile(name);
+        }
 
         // Locate the file
         // return fs.resolveFile(name.getPath());
-        return fs.resolveFile(name);
+        return testFile;
     }
 
     /**

